cloud/scope/powervs_cluster_test.go

```diff
@@ -1704,3 +1704,499 @@ func TestCreateServiceInstance(t *testing.T) {
 		g.Expect(err).To(BeNil())
 	})
 }
+
+func TestReconcileVPCSubnets(t *testing.T) {
+	var (
+		mockVPC  *mock.MockVpc
+		mockCtrl *gomock.Controller
+	)
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockVPC = mock.NewMockVpc(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When VPCSubnets are set in spec", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScope := PowerVSClusterScope{
+			IBMVPCClient: mockVPC,
+			IBMPowerVSCluster: &infrav1beta2.IBMPowerVSCluster{
+				ObjectMeta: metav1.ObjectMeta{Name: "ClusterName"},
+				Spec: infrav1beta2.IBMPowerVSClusterSpec{
+					VPCSubnets: []infrav1beta2.Subnet{{ID: ptr.To("subnet1ID"), Name: ptr.To("subnet1Name")}, {ID: ptr.To("subnet2ID"), Name: ptr.To("subnet2Name")}}},
+			},
+		}
+		subnet1Details := &vpcv1.Subnet{ID: ptr.To("subnet1ID"), Name: ptr.To("subnet1Name")}
+		subnet1Options := &vpcv1.GetSubnetOptions{ID: subnet1Details.ID}
+		mockVPC.EXPECT().GetSubnet(subnet1Options).Return(subnet1Details, nil, nil)
+		subnet2Details := &vpcv1.Subnet{ID: ptr.To("subnet2ID"), Name: ptr.To("subnet2Name")}
+		subnet2Options := &vpcv1.GetSubnetOptions{ID: subnet2Details.ID}
+		mockVPC.EXPECT().GetSubnet(subnet2Options).Return(subnet2Details, nil, nil)
+		instance, err := clusterScope.ReconcileVPCSubnets()
```

Comment:
```suggestion
		requeue, err := clusterScope.ReconcileVPCSubnets()
```
Please change wherever applicable